File: Ventas/cuPuntoVenta.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Ventas
{
	public partial class cuPuntoVenta : UserControl
	{
		Conexion con = new Conexion();
		DataTable dt = new DataTable();

		public cuPuntoVenta()
		{
			InitializeComponent();
			cbCliente.DataSource = con.clientes();
			cbCliente.DisplayMember = "nombre";
			cbCliente.ValueMember = "id";
		}

		private void tbCodigo_KeyDown(object sender, KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Enter)
			{
				dt = con.articulo(tbCodigo.Text);
				if (dt.Rows.Count != 0)
				{
					dgvArticulos.Rows.Add(dt.Rows[0]["id"], dt.Rows[0]["Descripcion"], dt.Rows[0]["Costo"], nudCantidad.Value);
					tbCodigo.Text = "";
					dt = null;
				}

			}
		}

		private void dgvArticulos_CellClick(object sender, DataGridViewCellEventArgs e)
		{
				if (e.ColumnIndex == 4)
				{
					dgvArticulos.Rows.Remove(dgvArticulos.SelectedRows[0]);
				}
		}

		private void dgvArticulos_RowsAdded(object sender, DataGridViewRowsAddedEventArgs e)
		{
			decimal total = 0;
			foreach (DataGridViewRow r in dgvArticulos.Rows)
			{
				total += Convert.ToDecimal(r.Cells["Costo"].Value) * Convert.ToDecimal(r.Cells["Cantidad"].Value);
			}

			tbImporte.Text = total.ToString();
		}

		private void dgvArticulos_RowsRemoved(object sender, DataGridViewRowsRemovedEventArgs e)
		{
			decimal total = 0;
			foreach (DataGridViewRow r in dgvArticulos.Rows)
			{
				total += Convert.ToDecimal(r.Cells["Costo"].Value)* Convert.ToDecimal(r.Cells["Cantidad"].Value);
			}

			tbImporte.Text = total.ToString();
		}

		private void cbCliente_DropDown(object sender, EventArgs e)
		{
			cbCliente.DataSource = con.clientes();
		}

		private void btnPagar_Click(object sender, EventArgs e)
		{
			con.registrarVenta(Convert.ToInt32(cbCliente.SelectedValue), Convert.ToDecimal(tbImporte.Text), global.id, dgvArticulos);
			dgvArticulos.Rows.Clear();
		}
	}
}

File: VentasVB/cuPuntoVenta.vb
Public Class cuPuntoVenta
    Dim con As Conexion = New Conexion()
    Dim dt As DataTable = New DataTable()
    Private Sub cuPuntoVenta_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        cbCliente.DataSource = con.clientes()
        cbCliente.DisplayMember = "nombre"
        cbCliente.ValueMember = "id"
    End Sub

    Private Sub cbCliente_DragDrop(sender As Object, e As DragEventArgs) Handles cbCliente.DragDrop
        cbCliente.DataSource = con.clientes()
    End Sub

    Private Sub tbCodigo_KeyDown(sender As Object, e As KeyEventArgs) Handles tbCodigo.KeyDown
        If e.KeyCode = Keys.Enter Then
            dt = con.articulo(tbCodigo.Text)
            If dt.Rows.Count <> 0 Then
                dgvArticulos.Rows.Add(dt.Rows(0)("id"), dt.Rows(0)("Descripcion"), dt.Rows(0)("Costo"), nudCantidad.Value)
                tbCodigo.Text = ""
                dt = Nothing
            End If

        End If
    End Sub

    Private Sub dgvArticulos_CellClick(sender As Object, e As DataGridViewCellEventArgs) Handles dgvArticulos.CellClick
        If e.ColumnIndex = 4 Then
            dgvArticulos.Rows.Remove(dgvArticulos.SelectedRows(0))
        End If
    End Sub

    Private Sub dgvArticulos_RowsAdded(sender As Object, e As DataGridViewRowsAddedEventArgs) Handles dgvArticulos.RowsAdded
        Dim total As Decimal = 0
        Dim r As DataGridViewRow
        For Each r In dgvArticulos.Rows
            total += Convert.ToDecimal(r.Cells("Costo").Value) * Convert.ToDecimal(r.Cells("Cantidad").Value)
        Next

        tbImporte.Text = total.ToString()
    End Sub

    Private Sub dgvArticulos_RowsRemoved(sender As Object, e As DataGridViewRowsRemovedEventArgs) Handles dgvArticulos.RowsRemoved
        Dim total As Decimal = 0
        Dim r As DataGridViewRow
        For Each r In dgvArticulos.Rows
            total += Convert.ToDecimal(r.Cells("Costo").Value) * Convert.ToDecimal(r.Cells("Cantidad").Value)
        Next

        tbImporte.Text = total.ToString()
    End Sub

    Private Sub btnPagar_Click(sender As Object, e As EventArgs) Handles btnPagar.Click
        con.registrarVenta(Convert.ToInt32(cbCliente.SelectedValue), Convert.ToDecimal(tbImporte.Text), idUsuario, dgvArticulos)
        dgvArticulos.Rows.Clear()
    End Sub
End Class
